src/is.test.ts

```diff
@@ -46,3 +47,97 @@ describe('isArray(value)', () => {
     expect(isArray(1)).to.equal(false);
   });
 });
+
+// isBoolean(value)
+describe('isBoolean(value)', () => {
+  it('Should be a function', () => {
+    expect(isBoolean).to.be.a('function');
+  });
+
+  it('Should return true when "value" is type boolean', () => {
+    expect(isBoolean(true)).to.equal(true);
+  });
+
+  it('Should return false when "value" is not type boolean', () => {
+    expect(isBoolean('This is a string')).to.equal(false);
+  });
+});
+
+// isInfinity(value)
+describe('isInfinity(value)', () => {
+  it('Should be a function', () => {
+    expect(isInfinity).to.be.a('function');
+  });
+
+  it('Should return true when "value" is Infinity', () => {
+    expect(isInfinity(Infinity)).to.equal(true);
+  });
+
+  it('Should return false when "value" is not Infinity', () => {
+    expect(isInfinity('This is a string')).to.equal(false);
+  });
+});
+
+// isInteger(value)
+describe('isInteger(value)', () => {
+  it('Should be a function', () => {
+    expect(isInteger).to.be.a('function');
+  });
+
+  it('Should return true when "value" is an Integer', () => {
+    expect(isInteger(24)).to.equal(true);
+  });
+
+  it('Should return false when "value" is not an Integer', () => {
+    expect(isInteger(24.245)).to.equal(false);
+  });
+});
+
+// isIntegerArray(value)
+describe('isInteger(value)', () => {
+  it('Should be a function', () => {
+    expect(isIntegerArray).to.be.a('function');
+  });
+
+  it('Should return true when "value" is type number[] and each array entry is an integer', () => {
+    expect(isIntegerArray([1, 2, 3])).to.equal(true);
+  });
+
+  it('Should return false when "value" is not type number[] or a number[] where each array entry is not an integer', () => {
+    expect(isIntegerArray([1, '2', true])).to.equal(false);
+  });
+
+  it('Should return false when "value" is not type number[] or a number[] where each array entry is not an integer', () => {
+    expect(isIntegerArray([1, 2.2, 3.3])).to.equal(false);
+  });
+});
```

Comment:
This can either be combined into a single `it` with multiple `expect`s, or the "should" message for each one can be changed to what specifically is being tested (preferred):

```suggestion
  it('Should return false when "value" is not type number[]', () => {
    expect(isIntegerArray([1, '2', true])).to.equal(false);
  });

  it('Should return false when "value" is type number[] but not every array entry is an integer', () => {
    expect(isIntegerArray([1, 2.2, 3.3])).to.equal(false);
  });
});
```

Reply:
Should I keep it to a single `it` to keep it inline with the first comment where we combined the falses into one it 

```
    expect(isBoolean('This is a string')).to.equal(false);
    expect(isBoolean(100)).to.equal(false);
    expect(isBoolean([])).to.equal(false);
    expect(isBoolean({})).to.equal(false);
 ```

Reply:
In this case, I think they might be better as separate tests. The reason is that the message for the first one is `Should return false when "value" is not type number[]`, so you could add multiple `expect`s that test other non-`number[]` values like `100` and `{}`. But the next test is for the case where the value is a `number[]`, but now we're looking at the specific type of number for each element.

Reply:
Agreed. That makes sense. In this case, we are trying to segregate the falses into a) when its not a number[] and b) to when its a number[] but not the type of number as elements that we are expecting.

Reply:
If we go with a two-`it` approach for every function tested, we would end up with messages like `Should return true when "value" is type number[] and each element is an integer` for one and `Should return false when "value" is not type number[] or when "value" is type number[] but each element is not an integer`, which is probably too long.